### ISSUE TITLE
PKCE with optional Client Secret

### ISSUE THREAD

**Issue by meyfa** [zitadel/zitadel#7110]
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

Opening this is a continuation of #6886, which was closed prematurely. I would still like this feature:

> Include an option to create a client secret to be used in conjunction with PKCE

### Describe your ideal solution

From #6886:

> In the UI one can have a checkbox, in Terraform and additional field, in api an additional parameter or boolean to define whether a client secret will be created or not when using PKCE

### Environment

Self-hosted

**Comment by brandan-schmitz** [comment]
I also agree this would be useful to have. I have encountered several applications that used PKCE but also required a client secret, some of which I currently have not used because I could not get them to work with Zitadel because of that missing client secret when using PKCE.

**Comment by hifabienne** [comment]
Thank you for sharing your idea.
If there is a significant demand from customers/community, we will carefully consider implementing the feature.
Currently, the issue will be added to our product backlog.

Meanwhile, if you're interested in implementing it yourself, we also welcome pull requests.

**Comment by sevensolutions** [comment]
Also waiting for this.
Here it says that PKCE is not a replacement for client authentication.

https://oauth.net/2/pkce/

**Comment by errnesto** [comment]
I think https://github.com/zitadel/oidc/pull/721 kind of solves this? We can not require PKCE but if we use it as a client it should be checked.

**Comment by sevensolutions** [comment]
@errnesto looks like, yes...
I quickly tested it using [traefik-oidc-auth](https://github.com/sevensolutions/traefik-oidc-auth) with PKCE enabled and using a client with authentication set to basic. ZITADEL was a managed instance.
At least from the client-side it looks ok. I think when i first tested this i received an error when i sent a code verifier and the client was not so set to "None" authentication.

But maybe someone of the ZITADEL team can confirm this?
